src/Extend/Middleware.php

```diff
@@ -21,40 +21,80 @@ class Middleware implements ExtenderInterface
     private $insertAfterMiddlewares = [];
     private $frontend;
 
+    /**
+     * @param string $frontend: The name of the frontend.
+     */
     public function __construct(string $frontend)
     {
         $this->frontend = $frontend;
     }
 
-    public function add($middleware)
+    /**
+     * Adds a new middleware to the frontend.
+     *
+     * @param string $middleware: ::class attribute of the middleware class.
+     *                            Must implement \Psr\Http\Server\MiddlewareInterface.
+     * @return self
+     */
+    public function add(string $middleware): self
     {
         $this->addMiddlewares[] = $middleware;
 
         return $this;
     }
 
-    public function replace($originalMiddleware, $newMiddleware)
+    /**
+     * Replaces an existing middleware of the frontend.
+     *
+     * @param string $originalMiddleware: ::class attribute of the original middleware class.
```

Comment:
Technically, it can also be a container binding name. That's probably true for multiple ::class parameters across all extenders, but even more so here because some middlewares are registered in the stack with their container name (like `flarum.api.route_resolver`)

I'm not sure if we want to mention this at all in the phpdoc here.

Reply:
Good point, I think it's worth mentioning it for original middleware, but not for new, because technically extensions shouldn't be binding middleware classes, at least not in normal circumstances.

Reply:
> technically extensions shouldn't be binding middleware classes

If people are advanced enough to use custom bindings, they should understand the IoC container well enough that they realize binding strings can be used instead of class attrs.